apps/docs/src/guide/types/arrays.md

```diff
@@ -1,20 +1,35 @@
 # Arrays
 
-## Passing in arrays
+In Sway, an `Array` is a fixed-size collection of elements of the same type, similar to a `Tuple`. `Arrays` can hold arbitrary types, including non-primitive types, with their size determined at compile time.
 
-You can pass a JavaScript `Array` into your contract method transparently, allowing passthrough of expected data via the type regardless of Array type. An Array in Sway is fixed in size, so maximum storage size is determined at time of program compile, not during runtime.
+## Using Arrays in the SDK
 
-An Array is just a Typed Array:
+You can pass a TypeScript `Array` into your contract method seamlessly, allowing the expected data to be passed through the type regardless of the `Array` type. In Sway, `Arrays` are fixed in size, so the storage size is determined at the time of program compilation, not during runtime.
 
-<!-- TODO: stop using hardcoded snippets -->
+An `Array` in Sway is simply a typed `Array`, as demonstrated in the following example:
 
-```ts:line-numbers
-// aka [u8; 2]
-let basicNumberArray = [1, 23];
-// aka [bool; 4]
-let basicBooleanArray = [true, false, true, false];
-```
+<<< @/../../docs-snippets/src/guide/types/arrays.test.ts#arrays-1{ts:line-numbers}
```

Comment:
Thoughts on adding the comment for the Sway comparison in the same line? Like this:
```ts
const numberArray: number[] = [1, 2, 3, 4, 5]; // in Sway: [u8; 5]
```